src/deriving.ml

```diff
@@ -676,6 +699,19 @@ let types_used_by_deriving (tds : type_declaration list) : structure_item list =
 let merge_generators field l =
   List.filter_map l ~f:(fun x -> x) |> List.concat |> Deriver.resolve_all field
 
+(* This function merges ['a derived] if they have the same [unused_code_warnings]. This
+   reduces the number of times we add [include struct ... end] to disable warnings. *)
+let merge_derived lists =
+  List.fold_right
```

Comment:
For some reason, I always thought `fold_left` and `fold_right` were the same function, with different order of arguments. So I was confused with this function. Finally, I learnt the right/left is about the traversal entry of the list!